### PR TITLE
[REVIEW] Add BlazingSQL version of query 28

### PR DESCRIPTION
**Pull request by Christian8491**
This PR adds the version of "query" 28 using `BlazingSQL`.

Contains: multiple functions from `tpcx_bb_query_28.py`

### CODE REVIEW

**Comment by Christian8491** [comment]
It shows **Correctness Assertion True** at scale factor **1K**, using **dgx201**.

```
(tpcx) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q28$ python tpcx_bb_query_28_sql.py --data_dir=/raid/tpcx-bb/sf1000/parquet_2gb/ --verify_results --verify_dir=/mnt/weka/nicholasb/tpcx-bb/validation/sf1000/ --cluster_host=10.150.162.27
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
[I] [23:34:04.065291] [<Future: finished, type: cuml.MultinomialNB, key: _fit-e511dff0-fb5a-4c15-b574-00d20a4efbc4>]
[I] [23:34:04.065698] [<Future: finished, type: cuml.MultinomialNB, key: _fit-538765af-48e5-41a7-b35d-457fd5e78628>]
[I] [23:34:04.065723] [<Future: finished, type: cuml.MultinomialNB, key: _fit-dd4e3609-900c-4caf-8db3-3a357e1895c5>]
[I] [23:34:04.065744] [<Future: finished, type: cuml.MultinomialNB, key: _fit-55086ab4-633e-4661-a3be-141361330e57>]
[I] [23:34:04.065764] [<Future: finished, type: cuml.MultinomialNB, key: _fit-32b30040-890a-42ae-97c3-9137cdcf83b6>]
[I] [23:34:04.065784] [<Future: finished, type: cuml.MultinomialNB, key: _fit-6e458481-6efe-4e50-9231-dc5340e00020>]
[I] [23:34:04.065804] [<Future: finished, type: cuml.MultinomialNB, key: _fit-5ec0223e-b69a-4361-95ff-8df7d8059db3>]
[I] [23:34:04.065822] [<Future: finished, type: cuml.MultinomialNB, key: _fit-2f752d5a-d69c-471a-9e5c-32cadb89b7a1>]
[I] [23:34:04.065840] [<Future: finished, type: cuml.MultinomialNB, key: _fit-602233c5-9fec-4ada-96c5-3d20393ef975>]
[I] [23:34:04.065859] [<Future: finished, type: cuml.MultinomialNB, key: _fit-4570bedd-9af2-479d-a86c-8f9e7710380c>]
[I] [23:34:04.065877] [<Future: finished, type: cuml.MultinomialNB, key: _fit-2b18ddb4-424e-4cea-bc21-45d0d4f131aa>]
[I] [23:34:04.065895] [<Future: finished, type: cuml.MultinomialNB, key: _fit-11d5e69a-7fa6-4c01-b898-7a9bdc01d7b8>]
[I] [23:34:04.065913] [<Future: finished, type: cuml.MultinomialNB, key: _fit-6ed79826-ab48-4c3f-9479-e7e8f1032947>]
[I] [23:34:04.065932] [<Future: finished, type: cuml.MultinomialNB, key: _fit-3bdab173-7986-4730-be57-763e744faaa3>]
[I] [23:34:04.065949] [<Future: finished, type: cuml.MultinomialNB, key: _fit-0806dd4a-4bbc-41cf-99ac-a95a4674f5a6>]
[I] [23:34:04.065967] [<Future: finished, type: cuml.MultinomialNB, key: _fit-52dd1683-6ff1-420b-9993-385e2d29eb6b>]
[I] [23:34:04.067471] [<Future: finished, type: cuml.MultinomialNB, key: _fit-e511dff0-fb5a-4c15-b574-00d20a4efbc4>]
[I] [23:34:04.067497] [<Future: finished, type: cuml.MultinomialNB, key: _fit-538765af-48e5-41a7-b35d-457fd5e78628>]
[I] [23:34:04.067518] [<Future: finished, type: cuml.MultinomialNB, key: _fit-dd4e3609-900c-4caf-8db3-3a357e1895c5>]
[I] [23:34:04.067538] [<Future: finished, type: cuml.MultinomialNB, key: _fit-55086ab4-633e-4661-a3be-141361330e57>]
[I] [23:34:04.067556] [<Future: finished, type: cuml.MultinomialNB, key: _fit-32b30040-890a-42ae-97c3-9137cdcf83b6>]
[I] [23:34:04.067575] [<Future: finished, type: cuml.MultinomialNB, key: _fit-6e458481-6efe-4e50-9231-dc5340e00020>]
[I] [23:34:04.067594] [<Future: finished, type: cuml.MultinomialNB, key: _fit-5ec0223e-b69a-4361-95ff-8df7d8059db3>]
[I] [23:34:04.067613] [<Future: finished, type: cuml.MultinomialNB, key: _fit-2f752d5a-d69c-471a-9e5c-32cadb89b7a1>]
[I] [23:34:04.067631] [<Future: finished, type: cuml.MultinomialNB, key: _fit-602233c5-9fec-4ada-96c5-3d20393ef975>]
[I] [23:34:04.067649] [<Future: finished, type: cuml.MultinomialNB, key: _fit-4570bedd-9af2-479d-a86c-8f9e7710380c>]
[I] [23:34:04.067667] [<Future: finished, type: cuml.MultinomialNB, key: _fit-2b18ddb4-424e-4cea-bc21-45d0d4f131aa>]
[I] [23:34:04.067685] [<Future: finished, type: cuml.MultinomialNB, key: _fit-11d5e69a-7fa6-4c01-b898-7a9bdc01d7b8>]
[I] [23:34:04.067703] [<Future: finished, type: cuml.MultinomialNB, key: _fit-6ed79826-ab48-4c3f-9479-e7e8f1032947>]
[I] [23:34:04.067721] [<Future: finished, type: cuml.MultinomialNB, key: _fit-3bdab173-7986-4730-be57-763e744faaa3>]
[I] [23:34:04.067739] [<Future: finished, type: cuml.MultinomialNB, key: _fit-0806dd4a-4bbc-41cf-99ac-a95a4674f5a6>]
[I] [23:34:04.067757] [<Future: finished, type: cuml.MultinomialNB, key: _fit-52dd1683-6ff1-420b-9993-385e2d29eb6b>]
[I] [23:34:04.070891] [Delayed('_merge_counts_to_model-41181b42-1fe4-4921-bf3a-2d8f0a0240df')]
Accuracy: 0.9985562602468817
Precision: 0.9973812100927781
Confusion Matrix: [[5.20630e+04 1.50000e+01 1.92000e+02]
 [6.10000e+01 3.54220e+04 1.30000e+02]
 [1.87000e+02 6.40000e+01 3.61393e+05]]
/home/christianc/miniconda3/envs/tpcx/lib/python3.7/site-packages/cudf/core/column/string.py:722: UserWarning: `n` parameter is not supported when                 `pat` and `repl` are list-like inputs
  `pat` and `repl` are list-like inputs"
Supervised Learning Query
Precisiom delta percent: 2.725112600221813
AUC delta percent: 2.846136696294336
Precision higher/similar: True
AUC higher/similar: True
Correctness Assertion True
Please install gspread and oauth2client               to use Google Sheets automation
```
NOTE: Is it completely normal that these kind of **<Future** messages appear?

**Comment by Christian8491** [comment]
Updated and tested again.

1. It shows **Correctness Assertion True** at scale factor **1K**, using **dgx201**.

```
(tpcx-bb) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q28$ python tpcx_bb_query_28_sql.py --config_file=/home/christianc/yaml_dir/validation_1k_config.yaml 
Using default arguments
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx-bb/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
[I] [00:50:42.945150] [<Future: finished, type: cuml.MultinomialNB, key: _fit-0953206f-f4fd-4989-818f-925f90cba2d9>]
[I] [00:50:42.945215] [<Future: finished, type: cuml.MultinomialNB, key: _fit-b57164b9-2c69-49c1-8ab0-82314aaf759f>]
[I] [00:50:42.945237] [<Future: finished, type: cuml.MultinomialNB, key: _fit-415dc16b-b255-40e2-8a43-c1312702b38d>]
[I] [00:50:42.945258] [<Future: finished, type: cuml.MultinomialNB, key: _fit-06e5bc80-8108-4565-bff2-4c71f4408052>]
[I] [00:50:42.945277] [<Future: finished, type: cuml.MultinomialNB, key: _fit-1f2fb431-4688-4af6-8fdf-6aa9da0b1a97>]
[I] [00:50:42.945297] [<Future: finished, type: cuml.MultinomialNB, key: _fit-c4f10957-b8d1-4ef8-a2ba-ad5fcc5e6182>]
[I] [00:50:42.945316] [<Future: finished, type: cuml.MultinomialNB, key: _fit-4be11d4c-d84a-43af-a984-03a3ea4e1c29>]
[I] [00:50:42.945334] [<Future: finished, type: cuml.MultinomialNB, key: _fit-546c328a-64df-4f34-a8d7-09dfed76e6ae>]
[I] [00:50:42.945352] [<Future: finished, type: cuml.MultinomialNB, key: _fit-470000bc-2c2e-4bc9-96f5-059a01be5141>]
[I] [00:50:42.945370] [<Future: finished, type: cuml.MultinomialNB, key: _fit-ee68bda5-3828-4dba-9460-33a2c6bd9385>]
[I] [00:50:42.945388] [<Future: finished, type: cuml.MultinomialNB, key: _fit-84c3ff4c-55a9-41fe-b7e2-e6157ba2dc23>]
[I] [00:50:42.945406] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a29d760d-1f1c-40fc-b958-a03c79e2c05e>]
[I] [00:50:42.945424] [<Future: finished, type: cuml.MultinomialNB, key: _fit-ac75a0d5-93bf-4da6-9d1d-5605df27e829>]
[I] [00:50:42.945442] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a79b95fc-4e4d-40da-90cf-e00f38bf0eea>]
[I] [00:50:42.945459] [<Future: finished, type: cuml.MultinomialNB, key: _fit-5190ab6c-b535-4a4e-8d88-1ce24d0157ae>]
[I] [00:50:42.945477] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a25a0ffb-935b-4627-82ee-471c185796bc>]
[I] [00:50:42.947092] [<Future: finished, type: cuml.MultinomialNB, key: _fit-0953206f-f4fd-4989-818f-925f90cba2d9>]
[I] [00:50:42.947120] [<Future: finished, type: cuml.MultinomialNB, key: _fit-b57164b9-2c69-49c1-8ab0-82314aaf759f>]
[I] [00:50:42.947141] [<Future: finished, type: cuml.MultinomialNB, key: _fit-415dc16b-b255-40e2-8a43-c1312702b38d>]
[I] [00:50:42.947161] [<Future: finished, type: cuml.MultinomialNB, key: _fit-06e5bc80-8108-4565-bff2-4c71f4408052>]
[I] [00:50:42.947180] [<Future: finished, type: cuml.MultinomialNB, key: _fit-1f2fb431-4688-4af6-8fdf-6aa9da0b1a97>]
[I] [00:50:42.947199] [<Future: finished, type: cuml.MultinomialNB, key: _fit-c4f10957-b8d1-4ef8-a2ba-ad5fcc5e6182>]
[I] [00:50:42.947217] [<Future: finished, type: cuml.MultinomialNB, key: _fit-4be11d4c-d84a-43af-a984-03a3ea4e1c29>]
[I] [00:50:42.947235] [<Future: finished, type: cuml.MultinomialNB, key: _fit-546c328a-64df-4f34-a8d7-09dfed76e6ae>]
[I] [00:50:42.947254] [<Future: finished, type: cuml.MultinomialNB, key: _fit-470000bc-2c2e-4bc9-96f5-059a01be5141>]
[I] [00:50:42.947272] [<Future: finished, type: cuml.MultinomialNB, key: _fit-ee68bda5-3828-4dba-9460-33a2c6bd9385>]
[I] [00:50:42.947291] [<Future: finished, type: cuml.MultinomialNB, key: _fit-84c3ff4c-55a9-41fe-b7e2-e6157ba2dc23>]
[I] [00:50:42.947309] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a29d760d-1f1c-40fc-b958-a03c79e2c05e>]
[I] [00:50:42.947327] [<Future: finished, type: cuml.MultinomialNB, key: _fit-ac75a0d5-93bf-4da6-9d1d-5605df27e829>]
[I] [00:50:42.947344] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a79b95fc-4e4d-40da-90cf-e00f38bf0eea>]
[I] [00:50:42.947362] [<Future: finished, type: cuml.MultinomialNB, key: _fit-5190ab6c-b535-4a4e-8d88-1ce24d0157ae>]
[I] [00:50:42.947380] [<Future: finished, type: cuml.MultinomialNB, key: _fit-a25a0ffb-935b-4627-82ee-471c185796bc>]
[I] [00:50:42.950732] [Delayed('_merge_counts_to_model-bb872fd4-1d79-4a88-9559-bcecd2387682')]
Accuracy: 0.9985562602468817
Precision: 0.9973812100927781
Confusion Matrix: [[5.20630e+04 1.50000e+01 1.92000e+02]
 [6.10000e+01 3.54220e+04 1.30000e+02]
 [1.87000e+02 6.40000e+01 3.61393e+05]]
/home/christianc/miniconda3/envs/tpcx-bb/lib/python3.7/site-packages/cudf/core/column/string.py:726: UserWarning: `n` parameter is not supported when                 `pat` and `repl` are list-like inputs
  `pat` and `repl` are list-like inputs"
Supervised Learning Query
Precisiom delta percent: 2.725112600221813
AUC delta percent: 2.846136696294336
Precision higher/similar: True
AUC higher/similar: True
Correctness Assertion True
Please install gspread and oauth2client               to use Google Sheets automation
```

from `validation_1k_config.yaml`, some useful values:

```
data_dir: /raid/tpcx-bb/sf1000/parquet_2gb/
verify_results: True
verify_dir: /mnt/weka/nicholasb/tpcx-bb/validation/sf1000/
```

2. The "query" finishes successfully at scale factor **10K**

```
(tpcx-bb) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q28$ python tpcx_bb_query_28_sql.py --config_file=/home/christianc/yaml_dir/succes_10k_config.yaml 
Using default arguments
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx-bb/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
[I] [00:56:08.493973] [Delayed('_merge_counts_to_model-94c11925-3de6-4bb3-b4f7-58da64164dbd')]
[I] [00:56:08.494709] [Delayed('_merge_counts_to_model-d013e601-056a-4815-8954-579540b7a4a8')]
[I] [00:56:08.495351] [Delayed('_merge_counts_to_model-a6ea9d40-af85-4279-a75c-492a7baa7673')]
[I] [00:56:08.496032] [Delayed('_merge_counts_to_model-c02b4f9c-0bcb-4d9a-9771-69a3eb3eeb32')]
[I] [00:56:08.496664] [Delayed('_merge_counts_to_model-6cd9fb69-48c8-44eb-bd10-e77f9a82d3e9')]
[I] [00:56:08.497299] [Delayed('_merge_counts_to_model-577f8c2a-3674-49c8-81e7-5790cb2cb0df')]
[I] [00:56:08.497937] [Delayed('_merge_counts_to_model-b8be544e-daca-4a84-98f6-112335ace336')]
[I] [00:56:08.498553] [Delayed('_merge_counts_to_model-1ba69bc6-e2ad-4296-bf9e-901823162b43')]
[I] [00:56:08.499169] [Delayed('_merge_counts_to_model-7ddde8c3-6868-4125-9752-c1868a1fe844')]
[I] [00:56:08.499791] [Delayed('_merge_counts_to_model-91af7b3f-be6f-49d4-927e-8efd294437e3')]
[I] [00:56:08.500407] [Delayed('_merge_counts_to_model-5bfc7dc1-74e1-41e6-994e-da0cf7794b1c')]
[I] [00:56:08.501022] [Delayed('_merge_counts_to_model-79694dfb-3d20-446c-aa34-0e79a2238c59')]
[I] [00:56:08.501644] [Delayed('_merge_counts_to_model-79fc877b-02e3-43a6-9165-0a1f1db87770')]
[I] [00:56:08.502272] [Delayed('_merge_counts_to_model-61438568-ee36-4a53-9add-41b66d45e5b5')]
[I] [00:56:08.502889] [Delayed('_merge_counts_to_model-8a7ce39e-e814-4e12-911e-3446f6145380')]
[I] [00:56:08.503506] [Delayed('_merge_counts_to_model-852a3e8c-ba36-4837-a230-76413d19caed')]
[I] [00:56:08.543399] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-94c11925-3de6-4bb3-b4f7-58da64164dbd>]
[I] [00:56:08.543438] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-d013e601-056a-4815-8954-579540b7a4a8>]
[I] [00:56:08.543461] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-a6ea9d40-af85-4279-a75c-492a7baa7673>]
[I] [00:56:08.543483] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-c02b4f9c-0bcb-4d9a-9771-69a3eb3eeb32>]
[I] [00:56:08.543504] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-6cd9fb69-48c8-44eb-bd10-e77f9a82d3e9>]
[I] [00:56:08.543524] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-577f8c2a-3674-49c8-81e7-5790cb2cb0df>]
[I] [00:56:08.543543] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-b8be544e-daca-4a84-98f6-112335ace336>]
[I] [00:56:08.543561] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-1ba69bc6-e2ad-4296-bf9e-901823162b43>]
[I] [00:56:08.543579] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-7ddde8c3-6868-4125-9752-c1868a1fe844>]
[I] [00:56:08.543598] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-91af7b3f-be6f-49d4-927e-8efd294437e3>]
[I] [00:56:08.543622] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-5bfc7dc1-74e1-41e6-994e-da0cf7794b1c>]
[I] [00:56:08.543640] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-79694dfb-3d20-446c-aa34-0e79a2238c59>]
[I] [00:56:08.543658] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-79fc877b-02e3-43a6-9165-0a1f1db87770>]
[I] [00:56:08.543676] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-61438568-ee36-4a53-9add-41b66d45e5b5>]
[I] [00:56:08.543694] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-8a7ce39e-e814-4e12-911e-3446f6145380>]
[I] [00:56:08.543713] [<Future: finished, type: cuml.MultinomialNB, key: _merge_counts_to_model-852a3e8c-ba36-4837-a230-76413d19caed>]
[I] [00:56:08.547021] [Delayed('_merge_counts_to_model-4c8afa3a-d08e-4e5c-bcb2-ea2ad6071d97')]
Accuracy: 0.9985007575777668
Precision: 0.9972272422789924
Confusion Matrix: [[3.036000e+05 1.260000e+02 1.163000e+03]
 [3.440000e+02 2.092550e+05 7.480000e+02]
 [1.223000e+03 3.530000e+02 2.122521e+06]]
/home/christianc/miniconda3/envs/tpcx-bb/lib/python3.7/site-packages/cudf/core/column/string.py:726: UserWarning: `n` parameter is not supported when                 `pat` and `repl` are list-like inputs
  `pat` and `repl` are list-like inputs"
Please install gspread and oauth2client               to use Google Sheets automation
```

from `succes_10k_config.yaml`, some useful values:

```
data_dir: /mnt/weka/tpcx-bb/sf10000/parquet_3gb/
verify_results: False
verify_dir: 
```

File: tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
+from xbb_tools.cluster_startup import attach_to_cluster
+from xbb_tools.text_vectorizers.dist_hashing_vectorizer import cudf_hashing_vectorizer
+import os
```
Comment (beckernick) [inline tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py:22]: Unused imports

File: tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+import cupy as cp
+import numpy as np
+
+from xbb_tools.utils import (
+    benchmark,
+    tpcxbb_argparser,
+    run_query,
+)
+
+cli_args = tpcxbb_argparser()
```
Comment (beckernick) [inline tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py:36]: I believe this is no longer needed

Comment (Christian8491) [reply tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py]: You are totally right.

File: tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py (hunk 1 above)
Comment (Christian8491) [reply tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py]: Removed.

**Pull request merged**